Opensegaapi/src/opensegaapi.cpp

```diff
@@ -1212,47 +1230,69 @@ extern "C" {
+		info("SEGAAPI_PlayWithSetup: hHandle: %08X dwNumSendRouteParams: %d pSendRouteParams: %08X dwNumSendLevelParams: %d pSendLevelParams: %08X dwNumVoiceParams: %d pVoiceParams: %08X dwNumSynthParams: %d pSynthParams: %08X", hHandle, dwNumSendRouteParams, *pSendRouteParams, dwNumSendLevelParams, *pSendLevelParams, dwNumVoiceParams, *pVoiceParams, dwNumSynthParams, *pSynthParams);
+		info("dwNumSynthParams: %d", dwNumSynthParams);
 #endif
 		OPEN_segaapiBuffer_t* buffer = (OPEN_segaapiBuffer_t*)hHandle;
+		buffer->playWithSetup = true;
 
 		for (int i = 0; i < dwNumSendRouteParams; i++)
 		{
-			SEGAAPI_SetSendRouting(hHandle, pSendRouteParams[i].dwChannel, pSendRouteParams[i].dwSend,
-				pSendRouteParams[i].dwDest);
+			SEGAAPI_SetSendRouting(hHandle, pSendRouteParams[i].dwChannel, pSendRouteParams[i].dwSend, pSendRouteParams[i].dwDest);
 		}
 
 		for (int i = 0; i < dwNumSendLevelParams; i++)
 		{
-			SEGAAPI_SetSendLevel(hHandle, pSendLevelParams[i].dwChannel, pSendLevelParams[i].dwSend,
-				pSendLevelParams[i].dwLevel);
+			SEGAAPI_SetSendLevel(hHandle, pSendLevelParams[i].dwChannel, pSendLevelParams[i].dwSend, pSendLevelParams[i].dwLevel);
 		}
 
+		unsigned int loopStart = 0;
+		unsigned int loopEnd = 0;
+		unsigned int loopState = 0;
+		unsigned int endOffset = 0;
+
 		for (int i = 0; i < dwNumVoiceParams; i++)
 		{
 			switch (pVoiceParams[i].VoiceIoctl)
 			{
 			case OPEN_VOICEIOCTL_SET_START_LOOP_OFFSET:
 				SEGAAPI_SetStartLoopOffset(hHandle, pVoiceParams[i].dwParam1);
+				loopStart = pVoiceParams[i].dwParam1;
 				break;
 			case OPEN_VOICEIOCTL_SET_END_LOOP_OFFSET:
 				SEGAAPI_SetEndLoopOffset(hHandle, pVoiceParams[i].dwParam1);
+				loopEnd = pVoiceParams[i].dwParam1;
 				break;
 			case OPEN_VOICEIOCTL_SET_END_OFFSET:
 				SEGAAPI_SetEndOffset(hHandle, pVoiceParams[i].dwParam1);
+				endOffset = pVoiceParams[i].dwParam1;
 				break;
 			case OPEN_VOICEIOCTL_SET_LOOP_STATE:
 				SEGAAPI_SetLoopState(hHandle, pVoiceParams[i].dwParam1);
+				loopState = pVoiceParams[i].dwParam1;
 				break;
 			case OPEN_VOICEIOCTL_SET_NOTIFICATION_POINT:
+#ifdef _DEBUG
+				info("Unimplemented! OPEN_VOICEIOCTL_SET_NOTIFICATION_POINT");
+#endif
 				break;
 			case OPEN_VOICEIOCTL_CLEAR_NOTIFICATION_POINT:
+#ifdef _DEBUG
+				info("Unimplemented! OPEN_VOICEIOCTL_CLEAR_NOTIFICATION_POINT");
+#endif
 				break;
 			case OPEN_VOICEIOCTL_SET_NOTIFICATION_FREQUENCY:
+#ifdef _DEBUG
+				info("Unimplemented! OPEN_VOICEIOCTL_SET_NOTIFICATION_FREQUENCY");
+#endif
 				break;
 			}
 		}
 
+#ifdef _DEBUG
+		info("loopdata\t%08X\t%08X\t%08X\t%08X\t%d\t%d", hHandle, loopStart, loopEnd, endOffset, loopState, buffer->size);
```

Comment:
change \t to space or something similar maybe

Reply:
ok, i used tabs at the time because i wanted easy import into excel ;)